server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -321,6 +330,60 @@ public ThreadPool(final Settings settings, final ExecutorBuilder<?>... customBui
         this.cachedTimeThread.start();
     }
 
+    private static ArrayList<Instrument> setupMetrics(MeterRegistry meterRegistry, String name, ExecutorHolder holder) {
+        Map<String, Object> at = Map.of();
+        ArrayList<Instrument> instruments = new ArrayList<>();
+        if (holder.executor() instanceof ThreadPoolExecutor threadPoolExecutor) {
+            String prefix = "es.thread_pool." + name + ".threads.";
+            logger.info("Starting tp metrics {}", name);
+            instruments.add(
+                meterRegistry.registerLongGauge(
+                    prefix + "count",
+                    "number of threads for " + name,
+                    "count",
```

Comment:
I know this feels odd to have a mandatory unit here, let's keep "count" here, we will look into this later

Reply:
An alternative I've seen applied elsewhere: use the object name, e.g. `{thread}`. See https://opentelemetry.io/docs/specs/semconv/general/metrics/#instrument-units
Just a suggestion, we do not have guidelines nor mandate anything.